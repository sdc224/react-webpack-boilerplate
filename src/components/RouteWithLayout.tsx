import React from "react";
import { Route, RouteProps } from "react-router-dom";

interface IProps {
	layout: React.FC;
}

const RouteWithLayout = (props: IProps & RouteProps) => {
	const { layout: Layout, component, ...rest } = props;
	const Component = component as React.ElementType;

	return (
		<Route
			{...rest}
			render={(matchProps) => (
				<Layout>
					<Component {...matchProps} />
				</Layout>
			)}
		/>
	);
};

export default RouteWithLayout;
