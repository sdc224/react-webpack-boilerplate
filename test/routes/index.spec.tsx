import React from "react";
import { render, RenderResult } from "@testing-library/react";
import RouteWithLayout from "@components/RouteWithLayout";
import MockComponent from "../mocks/MockComponent";
import { MemoryRouter, Switch, Redirect } from "react-router-dom";

let documentBody: RenderResult;

describe("<Route />", () => {
	beforeEach(() => {
		documentBody = render(
			<MemoryRouter>
				<Switch>
					<Redirect exact from="/" to="/mock" />
					<RouteWithLayout
						exact
						path="/mock"
						component={MockComponent}
						layout={MockComponent}
					/>
				</Switch>
			</MemoryRouter>
		);
	});
	it("has mock test", () => {
		const { getAllByText } = documentBody;
		expect(getAllByText(/Mock Component/i)).toHaveLength(2);
	});
	it("matches snapshot", () => {
		const { baseElement } = documentBody;
		expect(baseElement).toMatchSnapshot();
	});
});
