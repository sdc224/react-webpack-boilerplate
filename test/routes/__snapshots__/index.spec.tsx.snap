// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Route /> matches snapshot 1`] = `
<body>
  <div>
    <p>
      Mock Component
    </p>
    <p>
      Mock Component
    </p>
  </div>
</body>
`;
